WalletWasabi.Tests/UnitTests/StartWasabiOnSystemStartupTests.cs

```diff
@@ -18,8 +18,8 @@ public async Task ModifyStartupOnDifferentSystemsTestAsync()
 			}
 			else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
 			{
-				await Assert.ThrowsAsync<NotImplementedException>(async () => await StartupHelper.ModifyStartupSettingAsync(true));
-				await Assert.ThrowsAsync<NotImplementedException>(async () => await StartupHelper.ModifyStartupSettingAsync(false));
+				await StartupHelper.ModifyStartupSettingAsync(true);
+				await StartupHelper.ModifyStartupSettingAsync(false);
```

Comment:
Personally, I find these tests of little value. The only benefit is that you may find out that it either throws an exception or not. As a side-effect it breaks dev's settings for his/her installed WW.

Ideally, one would approach this test in a way that a test path is provided to where to write the data and then verify contents. Otherwise, the test tells you basically nothing and even has a negative side effect.

Reply:
Yes, the test is not the best. But we can do the write-to-file then verify approach only on the linux part of the test.
For Windows we need the read the registry and on OSX we need to run a command to get back the `Login Items`.
I will look into this and see if I can improve it or not.

Reply:
Yes, they are just throwing an exception and that is fine as these tests are basically testing the "real" function. I would not feel safe if these tests wouldn't be there. I value more these tests, taking them out is a no-go. 

Options:
- I was thinking about running them only in the CI, but it would be a new concept. 
- Let it be - devs can live with it.
- Check the original settings and restore unnecessary extra complexity.

I got it!! We can read back the original value and restore it from the UIConfig settings file.

Reply:
rant: Any *invalid modification* of the "tested" function that does not lead to an exception would pass the test. So even if the test passes, it tells you basically nothing. It certainly does not imply that the code is correct. Not sure why you find it useful then, but it seems you have made your mind.